### ISSUE TITLE
The terraform-provider-ibm_v1.64.0 plugin crashed!

### ISSUE THREAD

**Issue by ocofaigh** [IBM-Cloud/terraform-provider-ibm#5259]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform IBM Provider Version

terraform 1.5.7
ibm provider version 1.64.0

### Affected Resource(s)

* ibm_scc_profile_attachment

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

See https://github.com/terraform-ibm-modules/terraform-ibm-scc/pull/65
Logs at https://github.com/terraform-ibm-modules/terraform-ibm-scc/actions/runs/8542182407/job/23403425951

### Debug Output

```
│ Error: Plugin did not respond
│ 
│   with module.create_profile_attachment.ibm_scc_profile_attachment.scc_profile_attachment,
│   on ../../modules/attachment/main.tf line 40, in resource "ibm_scc_profile_attachment" "scc_profile_attachment":
│   40: resource "ibm_scc_profile_attachment" "scc_profile_attachment" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-ibm_v1.64.0 plugin:

panic: interface conversion: interface {} is *schema.Set, not []interface {}

goroutine 349 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc.resourceIbmSccProfileAttachmentMapToAttachmentsPrototype(0x409d100?)
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc/resource_ibm_scc_profile_attachment.go:637 +0x9d4
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc.resourceIbmSccProfileAttachmentMapToAttachmentPrototype(0x409d100?)
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc/resource_ibm_scc_profile_attachment.go:841 +0x13b
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc.resourceIbmSccProfileAttachmentCreate({0x5153d98, 0xc001f57e60}, 0x0?, {0x47eb140?, 0xc000ae0a80})
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc/resource_ibm_scc_profile_attachment.go:310 +0x71f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc001423180, {0x5153dd0, 0xc001e137d0}, 0xd?, {0x47eb140, 0xc000ae0a80})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:778 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001423180, {0x5153dd0, 0xc001e137d0}, 0xc0021ccc30, 0xc001e2a780, {0x47eb140, 0xc000ae0a80})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:909 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002ed488, {0x5153dd0?, 0xc001e136b0?}, 0xc001e34140)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000a62640, {0x5153dd0?, 0xc001e12ea0?}, 0xc0002c07e0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x46660e0?, 0xc000a62640}, {0x5153dd0, 0xc001e12ea0}, 0xc0002c0770, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000a521e0, {0x515ebe0, 0xc00097c4e0}, 0xc001e2c900, 0xc0014bcd50, 0x717ee98, 0x0)
	google.golang.org/grpc@v1.57.1/server.go:1358 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000a521e0, {0x515ebe0, 0xc00097c4e0}, 0xc001e2c900, 0x0)
	google.golang.org/grpc@v1.57.1/server.go:1735 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.57.1/server.go:970 +0xca
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.57.1/server.go:981 +0x15c

Error: The terraform-provider-ibm_v1.64.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by tyao117** [comment]
I identified the problem with code, and i'm working on putting out a release after.

**Comment by hkantare** [comment]
@tyao117 Can you let us know by when can we expect the PR
We are planning to release 1.64.1 by Monday 8th April

**Comment by ocofaigh** [comment]
Is there a test case that needs to be added too that would of caught this?

**Comment by pauljegouic** [comment]
Any update on this or any planning, cuz I need to know if we need to publish new modules with version <= 1.63 to avoid this error, or wait for you.
Those resource belongs to foundation module in my platform.

**Comment by tyao117** [comment]
I can add a fix by the end of this week.

**Comment by hkantare** [comment]
Fix is release as part of 1.64.1 version

https://registry.terraform.io/providers/IBM-Cloud/ibm/1.64.1
